src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageLoaderResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public struct PageLoaderResult
```

Comment:
We do have a couple of options here:

a) Pump the compilation change tokens to an `IActionDescriptorChangeProvider`. This seems difficult though given the layering.
b) We could add change tokens _ViewImports for all files as part of the `PageActionDescriptorChangeProvider`. It's not super bad since already do file traversals as part of the PageADP. Given that there's so there's so much overlap in the actual number of distinct change tokens, this seems a bit unnecessary.
c) We could add change tokens for _ViewImports for a (non-existent) file at Pages root. All the other _ViewImports are already tracked by the **/*.cshtml. 

Thoughts on the lot?